container/src/main/java/com/flipkart/poseidon/Poseidon.java

```diff
@@ -131,7 +131,11 @@ public void start() {
                 QueuedThreadPool threadPool = new QueuedThreadPool(jettyConfiguration.getMaxThreads(),
                         jettyConfiguration.getMinThreads(), jettyConfiguration.getThreadIdleTimeout(), queue);
                 server = new Server(threadPool);
-                ServerConnector connector = new ServerConnector(server, jettyConfiguration.getAcceptors(), jettyConfiguration.getSelectors());
+                ServerConnector connector = new ServerConnector(
+                        server,
+                        jettyConfiguration.getAcceptors(),
+                        jettyConfiguration.getSelectors(),
+                        new HttpConnectionFactory(Optional.of(jettyConfiguration.getHttpConfiguration()).orElseGet(HttpConfiguration::new)));
```

Comment:
If HttpConfiguration is provided, input buffer size won't be set to 16384. It will default to 8124 and there will be no way to override this. Instead HttpConnectionFactory itself can be taken from JettyConfiguration